src/3d/qgs3dutils.cpp

```diff
@@ -326,6 +326,18 @@ Qgs3DTypes::CullingMode Qgs3DUtils::cullingModeFromString( const QString &str )
     return Qgs3DTypes::NoCulling;
 }
 
+Qt3DRender::QCullFace::CullingMode Qgs3DUtils::qt3DCullingModeFromEnum( const Qgs3DTypes::CullingMode &mode )
+{
+  switch ( mode )
+  {
+    case Qgs3DTypes::NoCulling:    return Qt3DRender::QCullFace::NoCulling;
```

Comment:
It's not that easy. `Qgs3DTypes::CullingMode` is directly mapped to the index of the associated gui comboxbox whereas Qt3DRender::QCullFace has different values.

Reply:
you can attach user data to a combo box entry to handle this situation. It's much better then relying on a 1:1 map of enum to combo box indexes. See eg https://github.com/qgis/QGIS/blob/master/src/gui/symbology/qgssymbollayerwidget.cpp#L247

Reply:
So given this, can you address @benoitdm-oslandia's comment and remove the custom enum entirely?

Reply:
I tried to do so and it almost worked but I ended up with an error during sip generation in `QgsPolygon3DSymbol` for these methods: `cullingMode` and `setCullingMode`. `Qt3DRender::QCullFace::CullingMode` becomes exposed and I get this error: 

```
sip: Qt3DRender::QCullFace::CullingMode is undefined
```

I don't know if a dependency missing or some logic in sip but compilation fails.

Reply:
I'd just sip skip the methods -- we don't really support pyqgis for the majority of the 3d library

Reply:
Done.

Reply:
The culling mode enum (Qgs3DTypes::CullingMode) was introduced to allow to get/set culling mode in python, because it is unlikely that we would be supporting qt3d types anytime soon (would need to add pyqt3d as a dependency). No hard feelings if the enum is removed, but if someone will need to get/set culling mode on a 3d symbol in python, we will need to re-introduce this enum again...

Reply:
@wonder-sk that's fair enough -- I didn't realise it was already used elsewhere. In that case we should retain it, we can't remove without breaking api

Reply:
@wonder-sk and @nyalldawson Thanks for the comments. I have reverted the changes. `Qgs3DTypes::CullingMode` is kept.